realm/realm-library/src/objectServer/java/io/realm/SyncManager.java

```diff
@@ -48,6 +48,18 @@
 public class SyncManager {
 
     /**
+     * Debugging related options.
+     */
+    @SuppressFBWarnings("MS_SHOULD_BE_FINAL")
```

Comment:
So two different `BuildConfig` classes for `releases` and `debug`? One thing is user probably will have similar requirement as we do, to do testing on CI where the device doesn't have connections. So `BuildConfig` might not be a good place.